### ISSUE TITLE
Relax test for the Linux platform

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5489]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5489|
|Type|defect|
|Created|2012-02-16 21:04:40Z|

Here is a patch fixing the test for the Linux platform in t.i.test.test_unix ("linux2" is too specific: it could be "linux3" or even "linux" - in latest Python versions.)

Attachments:

* [linuxtest.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d49/d49784030ff04923c480ac7a800b03fedd67d329/e59ec19b9614a5a82b8cfda26dfed019b023ff22.patch) (1474 bytes) - added by antoine on 2012-02-16 21:04:53Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5489 5489
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1329426280000000 1329426280000000
changetime__1329665913000000 1329665913000000
version__None None
owner__exarkun exarkun
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks, Antoine.  Some of us still haven't realized we live in a world where "Linux 3" is a thing.

This change looks fine.  It does suggest to me that it is probably time we added `isLinux()` alongside the other platform-detection APIs in `twisted.python.runtime`.  Beyond these tests, I expect `twisted/internet/default.py` will soon want such a utility.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [33574]) Change Linux platform detection code to work with more versions of Linux

Author: antoine
Reviewer: exarkun
Fixes: [#5489](https://github.com/twisted/twisted/issues/5489)

Allow for the possibility of a "linux" or "linux3" platform identifier, in addition to
the more commonly seen "linux2".  These are all still Linux and should support
abstract namespace UNIX sockets.
